### DISCUSSION TITLE
allow referencing of page numbers in .pdf attachment file links

### DISCUSSION THREAD

**Discussion by freylax** [kaushalmodi/ox-hugo#656] [Ideas]
I would like to use org-links to local pdf documents located at an 'static' directory in the form of
[[file:static/myfile.pdf#page=7]]
currently this will not work because the .pdf extension is not separated from the page reference.
In emacs orgmode this also does not work but could probably solved by adding an 'pdf' hyperlink type
https://orgmode.org/manual/Adding-Hyperlink-Types.html
If citations could solve this issue I would like to know it.

**Comment by kaushalmodi** [comment]
> [[file:static/myfile.pdf#page=7]]

How are the links created in this form? Why not do `[[file:static/myfile.pdf]] page 7`?

> In emacs orgmode this also does not work

That's right. So I am curious how those links came up.

> but could probably solved by adding an 'pdf' hyperlink type

In that case, you can write your custom export function for the `pdf:` type as well. 

> If citations could solve this issue I would like to know it.

I don't use citations enough to understand how citations are related to pdf links.

> allow referencing of page numbers in .pdf attachment file links

How do you expect that to work with Hugo?

---

If you can share a working proof of concept in Markdown + Hugo, we can see if  `ox-hugo` can be bent to export that Markdown.

`ox-hugo` handles the Org to Markdown conversion, but before that, we need to find out what we need to convert to in Markdown. 😄

**Reply by freylax** [reply]
I can write this in markdown:
`[page 79, C Programming Language](/books/c-prog-lang.pdf#page=79)`
If you click the link you get directed to page 79 in the given pdf file.
By looking at the ox-hugo.el code it was not clear for me how the export mechanism for the pdf: would be invoked from ox-hugo. 
Also I would like to preserve the attachment magic with the static directory, thus the link would work from emacs and in the hugo generated pages.

**Reply by kaushalmodi** [reply]
As far as I know, Emacs doesn't natively support a link syntax for opening PDF files at specific pages. So I looked around if a solution already exists for supporting pdf page links, and I believe https://github.com/fuxialexander/org-pdftools comes close. That might work if you are already using `pdf-tools` to view PDF files in Emacs.

You can then advise `org-pdftools-export` to export to the format you prefer when exporting using `ox-hugo`.

**Reply by freylax** [reply]
Dear @kaushalmodi ,
I looked at org-pdftools and wrote the export for a "pdf" link type and "md" format using org-link-set-parameters. 
So I get the following result from org:
  `[[pdf:../static/books/c-prog-lang.pdf::79][page 79 in the C Programming Language]]`
  `[[file:../static/books/c-prog-lang.pdf::79][page 79 in the C Programming Language]]`
to md:
`[page 79 in the C Programming Language](../static/books/c-prog-lang.pdf#page=79)`
`[page 79 in the C Programming Language](/books/c-prog-lang.pdf)`
Thus the #page= part gets correctly translated, but the path is not handled equally like if I use the 'file' link type.
I argue that the 'file' export is handled differently in ox-hugo to tweak the path. Could we have a more general approach to include other link types for the 'static' dir handling as well?
Warm regards, Robert

**Reply by freylax** [reply]
I tweaked the path in my md export part now, this works fine.

**Reply by kaushalmodi** [reply]
@freylax I am glad you got this sorted out. I haven't yet found time to look into this.

Can you please share your full solution here? It can be helpful to someone also looking for a way to handle pdf links with page numbers.

**Reply by freylax** [reply]
Yes of course I will do so.
I added support for the LATeX backend too.
See here: [org-pdf.el](https://github.com/nodecum/nodecum-learning/blob/main/etc/org-pdf.el)
